modules/rte_mis_core/src/Plugin/Block/RoleBasedDetailsBlock.php

```diff
@@ -0,0 +1,599 @@
+                ($current_role == 'block_admin' && in_array($locationId, $locationIds))) {
+            $matchingSchools[] = $school;
+          }
+        }
+      }
+    }
+
+    return count($matchingSchools);
+  }
+
+  /**
+   * Gets the count of registered schools based on the user's role.
+   *
+   * @param string $current_role
+   *   The role of the user ('state_admin', 'district_admin', 'block_admin').
+   *
+   * @return int
+   *   The count of registered schools based on the user's role.
+   */
+  public function getRegisteredSchoolCount($current_role): int {
+    if ($current_role === 'state_admin') {
+      // Query for school admins.
+      $querySchoolAdmins = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('roles', 'school_admin')
+        ->condition('status', 1)
+        ->accessCheck(FALSE);
+      $school_admins_count = $querySchoolAdmins->count()->execute();
+
+      // Query for schools.
+      $querySchools = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('roles', 'school')
+        ->condition('status', 1)
+        ->accessCheck(FALSE);
+      $schools_count = $querySchools->count()->execute();
+
+      return $school_admins_count + $schools_count;
+    }
+
+    if ($current_role === 'district_admin' || $current_role === 'block_admin') {
+      return count($this->gettingMatchingSchools());
+    }
+
+    // Default return if $current_role doesn't match any condition.
+    return 0;
+  }
+
+  /**
+   * Gets the count of schools approved by a specific role.
+   *
+   * @param string $current_role
+   *   The role of the current user ('state_admin', 'district_admin',
+   *   'block_admin').
+   * @param string $role
+   *   The role for which approval status is checked ('District', 'Block', etc.)
+   *
+   * @return int
+   *   The count of schools approved by the specified role.
+   */
+  public function getSchoolApproveCount($current_role, $role): int {
+    $verification_status = 'school_registration_verification_approved_by_' . strtolower($role);
+
+    // Initialize variables.
+    $approved_count = 0;
+    $matchingSchoolIds = [];
+
+    if ($current_role === 'state_admin') {
+      // Step 1: Query all users with role 'school'.
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('roles', 'school')
+        ->condition('status', 1)
+        ->accessCheck(FALSE);
+      $school_ids = $query->execute();
+
+      // Return 0 if no schools found.
+      if (empty($school_ids)) {
+        return 0;
+      }
+
+      // Step 2: Filter schools by approval status.
+      $query = $this->entityTypeManager->getStorage('user')
+        ->getQuery()
+        ->condition('uid', $school_ids, 'IN')
+        ->accessCheck(FALSE)
+        ->condition('field_school_details.entity:mini_node.field_school_verification', $verification_status);
+      $approved_nids = $query->execute();
```

Comment:
- Can we combine these two query?
- Once the school user is approved by deo, the the user is granted with school_admin role and school is removed. So in this case the user with school roles will not exists.